lib/routes/user/received_invoice_notification.dart

```diff
@@ -38,15 +48,25 @@ class InvoiceNotificationsHandler {
         _setLoading(false);
         _handlingRequest = true;
 
+        // Do not pop dialog if there's a payment being processed
+        Future<bool> _onWillPop() async {
```

Comment:
I think it is cleaner to encapsulate the _inPogress in the PaymentRequestDialog class. It will move all the details level code including "onWillPop", "_sentPaymentResultSubscription" into the dialog itself and leave the logic here cleaner and more maintainable.
The dialog can also return status and we can use the result of the showDialog to show the appropriate flushbar.